### PR TITLE
Xx bug decimals in database

### PR DESCRIPTION
**Pull request by MathiasMeuleman**
Will close nothing really.

Bugfix to round floating-points to integers before logging to the database. Floating-points might occur when zooming in chrome.

### CODE REVIEW

**Comment by thervh70** [comment]
Hard to read method (`processRounding`), however this seems to work. Good job!

File: src/octopeer-github/content/EventTracker/MousePositionTracker.ts [hunk 1]
```diff
@@ -44,8 +44,8 @@ class MousePositionTracker implements EventTracker {
             const windowObject = $(window);
             const viewportX = windowObject.scrollLeft();
             const viewportY = windowObject.scrollTop();
-            this.db.post(EventFactory.mousePosition(viewportX + this.x, viewportY + this.y,
-                viewportX, viewportY), EMPTY_CALLBACK, EMPTY_CALLBACK);
+            this.db.post(EventFactory.mousePosition(Math.round(viewportX + this.x), Math.round(viewportY + this.y),
+                Math.round(viewportX), Math.round(viewportY)), EMPTY_CALLBACK, EMPTY_CALLBACK);
```
Comment (mpsijm) [inline src/octopeer-github/content/EventTracker/MousePositionTracker.ts]: Why also round here when the coordinates are also rounded in the `RESTApiDatabaseAdapter`?

File: src/octopeer-github/main/Database/RESTApiDatabaseAdapter.ts [hunk 2]
```diff
@@ -86,11 +86,28 @@ class RESTApiDatabaseAdapter implements DatabaseAdaptable {
...
+     * Round all necessary fields to integers. Necessary to stay compatible with the database.
+     * @param eventData The data to be send to the database.
+     */
+    private processRounding(eventData: EventObjectData): any {
+        const roundCandidates: string[] = ["position_x", "position_y", "viewport_x", "viewport_y"];
+        let candidate: string;
+        const indexableEventData = <{[name: string]: number; }>(<any>eventData);
+        for (let i = 0; i < roundCandidates.length; i++) {
+            candidate = roundCandidates[i];
+            if ([candidate] !== undefined) {
```
Comment (mpsijm) [inline src/octopeer-github/main/Database/RESTApiDatabaseAdapter.ts]: Is this if-condition needed? An array that contains the variable `candidate` is always defined.

File: src/octopeer-github/content/EventTracker/MousePositionTracker.ts (hunk 1 above)
Comment (MathiasMeuleman) [inline src/octopeer-github/content/EventTracker/MousePositionTracker.ts]: old code, my mistake

File: src/octopeer-github/main/Database/RESTApiDatabaseAdapter.ts (hunk 2 above)
Comment (MathiasMeuleman) [inline src/octopeer-github/main/Database/RESTApiDatabaseAdapter.ts]: missed something there... should be `if (indexableEventData[candidate]...)`. Is fixed

**Comment by MathiasMeuleman** [comment]
I have fixed all feedback, also added additional explanatory TSDoc to the processRounding function

**Comment by mpsijm** [comment]
Looks good, at the moment I don't have time to try out the functionality, but if others confirm that it works, this can be merged. :)

File: src/octopeer-github/main/Database/RESTApiDatabaseAdapter.ts [hunk 3]
```diff
@@ -86,11 +86,30 @@ class RESTApiDatabaseAdapter implements DatabaseAdaptable {
...
         }
     }
 
     /**
+     * Round all necessary fields to integers. Necessary to stay compatible with the database.
+     * A list of all items that can occur in an EventObjectData is given. For each of these 'candidates'
+     * it is checked whether it is contained in the given EventDataObject and if so, rounded.
+     * @param eventData The data to be send to the database.
+     */
+    private processRounding(eventData: EventObjectData): any {
```
Comment (mdingjan) [inline src/octopeer-github/main/Database/RESTApiDatabaseAdapter.ts]: This function has been tested by magic. :innocent:

Comment (MathiasMeuleman) [inline src/octopeer-github/main/Database/RESTApiDatabaseAdapter.ts]: But of course! And how do you expect me to test a private function?

Comment (mdingjan) [inline src/octopeer-github/main/Database/RESTApiDatabaseAdapter.ts]: Via a public one xD
But it has been covered, so you won't hear me 'mocking' ;)

**Comment by mdingjan** [comment]
When zoomed in, I still get 400s on mouse-position-events and mouse-scroll-events.
Can you have one more look at this and try to reproduce it?

**Comment by mdingjan** [comment]
So I tried to reproduce the 400s again, rebuilding & deleting the old build from both the browser and the directory, but didn't succeed. 
The implementation does work as it is supposed to.

Maybe one more person can build the extension on this PR and check it, just to be sure?

File: src/octopeer-github/test/main/Database/RESTApiDatabaseAdapterTest.ts [hunk 4]
```diff
@@ -135,4 +135,17 @@ describe("A RESTApiDatabaseAdapter", function() {
...
         expect(consoleSpy.calls.all()[0].args[0]).toBe("Constructed DatabaseAdapter(http://localhost:8000/)");
     });
+
+    it("has to round data that is only accepted as integers", function() {
+        let joc = jasmine.objectContaining;
+
+        adapter.post(EventFactory.mouseScroll(23.1, 12.8, 123.456), spyFunc, EMPTY_CALLBACK);
+        let expected = (<any>jasmine.Ajax.requests.mostRecent()).params;
+
+        expect(JSON.parse(expected)).toEqual(joc({
```
Comment (mpsijm) [inline src/octopeer-github/test/main/Database/RESTApiDatabaseAdapterTest.ts]: Do we still need the `joc` alias or can you just turn this back into `jasmine.objectContaining`?

**Comment by mpsijm** [comment]
@mdingjan I don't see any troubles, but like you said to us in the chat you don't see any problems anymore either. :)
@MathiasMeuleman I like the new test, thanks a lot :) Also thanks for quickly resolving my feedback, can now be merged if others agree :)

**Comment by mdingjan** [comment]
I agree, thanks for processing the feedback. :)

**Comment by Exclaminator** [comment]
LGTM, merging.

**Pull request merged**